packages/protocol/contracts/governance/Validators.sol

```diff
@@ -1131,4 +1131,29 @@ contract Validators is
       }
     }
   }
+
+  /**
+  * @notice Returns the group that `account` was a member of during `epochNumber`.
+  * @param account The account whose group membership should be returned.
+  * @param epochNumber The epoch number we are querying this account's membership at.
+  * @param index The index into the validator's history struct for their history at `epochNumber`.
+  * @return The group that `account` was a member of during `epochNumber`.
+  */
+  function groupMembershipInEpoch(address account, uint256 epochNumber, uint256 index)
+    external
+    view
+    onlyRegisteredContracts(canForceDeaffiliation)
+    returns (address)
+  {
+    require(isValidator(account) && epochNumber <= getEpochNumber());
+    MembershipHistory storage history = validators[account].membershipHistory;
+    require(
+      history.entries[index].epochNumber == epochNumber ||
+        (history.entries[index].epochNumber < epochNumber &&
+          (history.entries[index.add(1)].epochNumber > epochNumber ||
+            index == history.numEntries.sub(1))),
```

Comment:
I don't think this is how you check that this is the last entry in `history` (though I agree that's what you want to check)

Reply:
Right, it should be `index.sub(history.tail) == history.numEntries.sub(1)` since the tail is the smallest index, so `index` - `tail` should give the offset in entries, which if equal to the total entries means we are on the last element. Good catch